server/cmwell-ws/app/security/httpauth/BasicHttpAuthentication.scala

```diff
@@ -23,15 +23,12 @@ import scala.util.{Failure, Success, Try}
 trait BasicHttpAuthentication {
   private val authTypePrefix = "Basic "
 
-  def decodeBasicAuth(auth: String): Try[(String, String)] = {
-    Try(new String(new Base64().decode(auth.drop(authTypePrefix.length)), "UTF-8")) match {
-      case Success(userColonPass) =>
-        val split = userColonPass.split(":", 2)
-        if (split.length != 2)
-          Failure(new IllegalArgumentException(s"$auth is not a valid Basic header"))
-        else
-          Success(split(0)->split(1))
-      case Failure(_) => Failure(new IllegalArgumentException(s"$auth is not a valid Basic header"))
-    }
+  def decodeBasicAuth(auth: String): (String, String) = {
+    val userColonPass = new String(new Base64().decode(auth.drop(authTypePrefix.length)), "UTF-8")
+    val split = userColonPass.split(":", 2)
```

Comment:
Might be better to use `cmwell.util.string.splitAtNoSep`

Reply:
I disagree. I need to validate the input was in form of `base64(user:pass)` (i.e. after invoking decode, that the split returned an exact length of 2 elements). `splitAtNoSep` is too forgiving (e.g. it returns `(str,"")` if separator does not exist in input. I don't want that behavior here, or I will return 401 even for 400 cases.